libshortfin/src/shortfin/local/async.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 Advanced Micro Devices, Inc
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "shortfin/local/async.h"
+
+namespace shortfin::local {
+
+CompletionEvent::CompletionEvent()
+    : wait_source_(iree_wait_source_immediate()) {}
+
+CompletionEvent::CompletionEvent(iree::shared_event::ref event)
+    : wait_source_(event->await()) {
+  // // It is sufficient to simply capture the event ref in the lambda. It
```

Comment:
TODO?

Reply:
Oh, thanks for the catch. Refactor that I didn't finish. Meant to initialize the resource_baton (was using a naked lambda for this, which is where the comment came from).